cargo-concordium/src/build.rs

```diff
@@ -375,6 +384,237 @@ pub fn init_concordium_project(path: impl AsRef<Path>) -> anyhow::Result<()> {
     Ok(())
 }
 
+/// Converts the ContractV0 schema of the given contract_name to JSON and writes
+/// it to a file named after the smart contract name at the specified location.
+pub fn write_json_schema_to_file_v0(
+    absolute_path_of_out: &Path,
+    contract_name: &str,
+    contract_schema: &ContractV0,
+) -> anyhow::Result<()> {
+    // create empty schema_json
+    let mut schema_json: Value = Value::Object(serde_json::Map::new());
+
+    // add init schema
+    if let Some(init_schema) = &contract_schema.init {
+        schema_json["init"] = type_to_json(init_schema);
+    }
+
+    // add state schema
+    if let Some(state_schema) = &contract_schema.state {
+        schema_json["state"] = type_to_json(state_schema);
+    }
+
+    // add receive entrypoints
+    if !contract_schema.receive.is_empty() {
+        // create empty entrypoints
+        let mut entrypoints: Value = Value::Object(serde_json::Map::new());
+
+        // iterate through the entrypoints and add their schemas
+        for (method_name, receive_schema) in contract_schema.receive.iter() {
+            // add `method_name` entrypoint
+            entrypoints[method_name] = type_to_json(receive_schema);
+        }
+
+        // add all receive entrypoints
+        schema_json["entrypoints"] = entrypoints;
+    }
+
+    // save the schema JSON representation into the file.
+    let mut out_path = absolute_path_of_out.to_owned();
+    out_path.push(contract_name.to_owned() + "_schema.json");
```

Comment:
Consider sanitizing the contract name, since a contract name might contain `/` or characters which are not allowed in file names.

Reply:
only ascii alphanumeric and `"-_[]{}"` are now allowed.